### PR TITLE
[Pool Simulator] Enable Analysis Page pt.1

### PR DESCRIPTION
**Pull request by luizakp**
This PR enables the analysis page to:
- metastable vs metastable
- Gyro E vs Gyro E
- Gyro E vs Metastable

I do not enable for:
- Metastable vs Gyro E

https://www.loom.com/share/c2dc37a2907d49d28d1a465039f92007?sid=60a9a030-64ca-42d6-8eb3-a4c2412478cb

### CODE REVIEW

File: apps/balancer-tools/src/app/poolsimulator/(components)/Menu.tsx [hunk 1]
```diff
@@ -231,7 +242,7 @@ export default function Menu() {
     ) {
       return <Spinner />;
     }
-    return <div className="text-slate12">Analysis Menu</div>;
+    return <AnalysisPoolParamsForm />;
```
Comment (yvesfracari) [inline apps/balancer-tools/src/app/poolsimulator/(components)/Menu.tsx:245]: Should we make the same pattern and let both `AnalysisPoolParamsForm` and `IndexMenu` components in or outside the `Menu` file?

File: apps/balancer-tools/src/app/poolsimulator/(utils)/index.ts [hunk 2]
```diff
@@ -9,15 +9,15 @@ import { PoolTypeEnum } from "../(types)";
 
 export function convertAnalysisDataToAMM(data: AnalysisData) {
   if (!data.poolType) return;
-
+  if (!data.poolParams?.swapFee) return;
```
Comment (yvesfracari) [inline apps/balancer-tools/src/app/poolsimulator/(utils)/index.ts:12]: would merge those 2 conditions

File: packages/math-poolsimulator/src/index.ts [hunk 3]
```diff
@@ -15,7 +15,9 @@ export class AMM<TPoolPairData extends PoolPairData> {
     tokenOut: string,
   ): number {
     const poolPairData = this.math.parsePoolPairData(tokenIn, tokenOut);
-    return this.math._exactTokenInForTokenOut(poolPairData, amountIn);
+    return this.math
+      ._exactTokenInForTokenOut(poolPairData, bnum(amountIn))
+      .toNumber();
```
Comment (yvesfracari) [inline packages/math-poolsimulator/src/index.ts:20]: 👍

**Review by yvesfracari — APPROVED** [review]
These prettier diffs are really annoying to review hahaha. Just let some minor comments but we are good to go after this.

File: apps/balancer-tools/src/app/poolsimulator/(components)/DepthCost.tsx [hunk 4]
```diff
@@ -132,13 +154,14 @@ export function DepthCost() {
   };
 
   return <Plot {...props} />;
+  return <div className="text-slate12">depth cost</div>;
```
[the remark this answers is not in the document]
Comment (luizakp) [reply apps/balancer-tools/src/app/poolsimulator/(components)/DepthCost.tsx]: ops

File: apps/balancer-tools/src/app/poolsimulator/(components)/Menu.tsx (hunk 1 above)
Comment (luizakp) [reply apps/balancer-tools/src/app/poolsimulator/(components)/Menu.tsx:245]: I think we can discuss this better on https://linear.app/bleu-llc/issue/BAL-548/enable-analysis-page-with-dynamic-data but I would do both outside of `Menu`

File: apps/balancer-tools/src/app/poolsimulator/(components)/DepthCost.tsx [hunk 5]
```diff
@@ -199,7 +222,11 @@ function calculateDepthCostAmount(
 
   const currentSpotPrice = amm.spotPrice(tokenIn.symbol, tokenOut.symbol);
 
-  const newSpotPrice = currentSpotPrice * 1.02;
+  //TODO fix this on BAL-503
```
Comment (ribeirojose) [inline apps/balancer-tools/src/app/poolsimulator/(components)/DepthCost.tsx:225]: 👍

File: apps/balancer-tools/src/app/poolsimulator/(components)/AnalysisPoolParamsForm.tsx [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+  const {
+    setValue,
+    getValues,
+    formState: { errors },
+  } = form;
+
+  const onSubmit = () => {
+    if (Object.keys(errors).length) return;
+  };
```
Comment (ribeirojose) [inline apps/balancer-tools/src/app/poolsimulator/(components)/AnalysisPoolParamsForm.tsx:31]: 👀 there's something missing here, no?

File: apps/balancer-tools/src/app/poolsimulator/(components)/AnalysisPoolParamsForm.tsx [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+  useEffect(() => {
+    setAnalysisTokenByIndex(0);
+    setCurrentTabTokenByIndex(1);
```
Comment (ribeirojose) [inline apps/balancer-tools/src/app/poolsimulator/(components)/AnalysisPoolParamsForm.tsx:35]: Shouldn't these be the default?

File: apps/balancer-tools/src/app/poolsimulator/(components)/AnalysisPoolParamsForm.tsx (hunk 6 above)
Comment (luizakp) [reply apps/balancer-tools/src/app/poolsimulator/(components)/AnalysisPoolParamsForm.tsx:31]: not really... I just meant to add the sidebar with the token selector and used what we already had on StableSwap. But for this use case, we don't even need a form rn 😆 will remove it

File: apps/balancer-tools/src/app/poolsimulator/(components)/AnalysisPoolParamsForm.tsx (hunk 7 above)
Comment (luizakp) [reply apps/balancer-tools/src/app/poolsimulator/(components)/AnalysisPoolParamsForm.tsx:35]: `setAnalysisTokenByIndex()` is not a normal state function but this:

```
 function setAnalysisTokenByIndex(index: number) {
    const token = initialData.tokens[index];
    if (token) setAnalysisToken(token);
  }
```

And `setAnalysisToken()` by default is not the data we imported it. Does that make sense to you?

**Pull request merged**